plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -16,10 +17,183 @@
  * @since 9.2.0
  * @package WooCommerce\Classes
  */
-class RemoteLogger {
+class RemoteLogger extends \WC_Log_Handler {
+	const LOG_ENDPOINT                = 'https://public-api.wordpress.com/rest/v1.1/logstash';
+	const THROTTLE_TRANSIENT          = 'woocommerce_remote_logging_throttle';
+	const THROTTLE_LIMIT              = 10; // Maximum number of requests in the interval.
+	const THROTTLE_INTERVAL           = 300; // Throttle interval in seconds (5 minutes).
 	const WC_LATEST_VERSION_TRANSIENT = 'latest_woocommerce_version';
 	const FETCH_LATEST_VERSION_RETRY  = 'fetch_latest_woocommerce_version_retry';
 
+	/**
+	 * The logger instance.
+	 *
+	 * @var \WC_Logger_Interface|null
+	 */
+	private $local_logger;
+
+	/**
+	 * Remote logger constructor.
+	 *
+	 * @internal
+	 * @param \WC_Logger_Interface|null $logger Logger instance.
+	 */
+	public function __construct( \WC_Logger_Interface $logger = null ) {
+		if ( null === $logger ) {
+			$this->local_logger = wc_get_logger();
+		}
+		$this->local_logger = $logger;
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int    $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug.
+	 * @param string $message Log message.
+	 * @param array  $context Additional information for log handlers.
+	 *
+	 * @throws \Exception If the remote logging fails. The error is caught and logged locally.
+	 *
+	 * @return bool False if value was not handled and true if value was handled.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
+		if ( ! \WC_Log_Levels::is_valid_level( $level ) ) {
+			/* translators: 1: WC_Remote_Logger::log 2: level */
+			wc_doing_it_wrong( __METHOD__, sprintf( __( '%1$s was called with an invalid level "%2$s".', 'woocommerce' ), '<code>WC_Remote_Logger::log</code>', $level ), '9.2.0' );
```

Comment:
I decided to keep it as is for now because WC Logger also calls `wc_doing_it_wrong()` with the method name in the message.  I think it would be better to keep it consistent.

https://github.com/woocommerce/woocommerce/blob/f3883053f3a50689c034450ba28de6b996d8eb7a/plugins/woocommerce/includes/class-wc-logger.php#L168-L171

Reply:
Updated it to `::handle()` in f3883053f3a50689c034450ba28de6b996d8eb7a